pack/pack.go

```diff
@@ -52,6 +55,18 @@ func (p *Pack) generateId() {
 	p.Id = id
 }
 
+func (p *Pack) SetStatus() {
```

Comment:
It looks weird to me to need to call this method in order to update the status when you already have the information within the struct. I don't know if adding a `MarshalJSON` method and decorating with a Status field would be cleaner. It will also avoid problems if we forget to add that call to `SetStatus` in the future.

```
func (p *Pack) MarshalJSON() ([]byte, error) {
	return json.Marshal(&struct {
		Status string `json:"status"`
		*Pack
	}{
		status: lastSeenToStatus(),
		Pack:   p,
	})
}
```

Reply:
Yes it could be done like this however if this won't be JSON then it would require another method. Apart from that this would make status field hidden.

I don't personally like that I have to set field. Ideally it would be `getStatus` etc... and marshaller would take care of this.